packages/playwright/src/runner/runner.ts

```diff
@@ -152,17 +152,24 @@ export class Runner {
 export type LastRunInfo = {
   status: FullResult['status'];
   failedTests: string[];
+  testDurations?: { [testId: string]: number };
 };
 
 async function writeLastRunInfo(testRun: TestRun, status: FullResult['status']) {
   const [project] = filterProjects(testRun.config.projects, testRun.config.cliProjectFilter);
   if (!project)
     return;
   const outputDir = project.project.outputDir;
-  await fs.promises.mkdir(outputDir, { recursive: true });
-  const lastRunReportFile = path.join(outputDir, '.last-run.json');
+  const lastRunReportFile = testRun.config.lastRunFile || path.join(outputDir, '.last-run.json');
+  const lastRunReportFileDir = path.dirname(lastRunReportFile);
+  await fs.promises.mkdir(lastRunReportFileDir, { recursive: true });
   const failedTests = testRun.rootSuite?.allTests().filter(t => !t.ok()).map(t => t.id);
-  const lastRunReport = JSON.stringify({ status, failedTests }, undefined, 2);
+  const testDurations = testRun.rootSuite?.allTests().reduce((map, t) => {
+    if (t.results.length)
+      map[t.id] = t.results.reduce((a, b) => a + b.duration, 0);
+    return map;
+  }, {} as { [testId: string]: number });
+  const lastRunReport = JSON.stringify({ status, failedTests, testDurations }, undefined, 2);
```

Comment:
The `lastrun` reporter is required to generate last run info when merging blob reports from shards.

Maybe it would be better to implicitly write the lastrun info when merging blob reports?

Reply:
I think so. From the user's perspective, lastrun just working out of the box would be easier. Whether it is a reporter internally or not is an implementation detail.

Reply:
How about always using the last run reporter? Then we would only need to have a single implementation….

Reply:
That definitely works. We'll have to add it [here](https://github.com/microsoft/playwright/blob/main/packages/playwright/src/runner/reporters.ts#L37) when `isTestServer === false`.